src/sortStudents.ts

```diff
@@ -1,16 +1,70 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(grades: number[]): number {
+  return grades.reduce((sum, value) => sum + value, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        return sortedStudents.sort((a, b) => {
+          return a[sortBy].localeCompare(b[sortBy]);
+        });
+      }
+
+      return sortedStudents.sort((a, b) => {
+        return b[sortBy].localeCompare(a[sortBy]);
+      });
+
+    case SortType.Age:
+    case SortType.Married:
+      if (order === 'asc') {
+        return sortedStudents.sort((a, b) => {
+          return +a[sortBy] - +b[sortBy];
```

Comment:
```suggestion
          return Number(a[sortBy]) - Number(b[sortBy]);
```
Number() is more obvious way